### PR TITLE
Make forward options optional

### PR DESCRIPTION
**Pull request by omegas27**
Towards https://github.com/giantswarm/giantswarm/issues/10460

### CODE REVIEW

File: helm/coredns-app/values.yaml [hunk 1]
```diff
@@ -20,10 +20,11 @@ configmap:
   log: |
     denial
     error
-  # Uncoment forward and forwadOptions to enable DNS forward https://coredns.io/plugins/forward/
+
+  # Uncoment forward and forwardOptions to enable DNS forward https://coredns.io/plugins/forward/
```
Comment (sslavic) [inline helm/coredns-app/values.yaml:24]:
```suggestion
  # Uncomment and define `forward` to enable and configure forwarding rules for CoreDNS forward plugin (for syntax see https://coredns.io/plugins/forward/)
  # Default `forwardOptions` are defined separately below; these are applied and required when forward plugin is enabled.
```

File: helm/coredns-app/values.yaml [hunk 2]
```diff
@@ -20,10 +20,11 @@ configmap:
...
   # forward: . 192.168.1.1 /etc/resolv.conf
-  # forwardOptions: |
-  #   policy sequential
+  forwardOptions: |
+    policy sequential
```
Comment (sslavic) [inline helm/coredns-app/values.yaml:28]: Not competent to judge, just curious do we have rationale for having forward options different from default upstream policy of `policy random`? If we do, maybe we should leave a comment here explaining that rationale, even if just for us.

Comment (omegas27) [reply helm/coredns-app/values.yaml:28]: https://github.com/giantswarm/kubernetes-coredns/pull/27/files#r271379790

Comment (omegas27) [reply helm/coredns-app/values.yaml:28]: It was to ensure the order but I am not sure round_robin does not make more sense

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project's packages adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [v1.1.9] 2020-05-04
```
Comment (sslavic) [inline CHANGELOG.md:8]: We kind of agreed to have release preparation in separate PR. @kopiczko suggests rationale for that is to make reverting changes easier. For that purpose probably even adding entry to Unreleased changelog update should be included in a separate PR from the one making the actual change, as reverting change commit should probably not revert changelog entry.

Comment (omegas27) [reply CHANGELOG.md:8]: Let me split that up. Would 1 PR for the change and one PR for the release notes do?

File: helm/coredns-app/values.yaml (hunk 2 above)
Comment (sslavic) [reply helm/coredns-app/values.yaml:28]: @pipo02mix does that rationale (comment Quentin referred to) still hold? If so, Quentin please add it as comment in values.yaml that upstream default is policy random but we think policy sequential is batter default due to ...

File: CHANGELOG.md (hunk 3 above)
Comment (xh3b4sd) [reply CHANGELOG.md:8]: I do not really follow this here. Too many information for me and I am not into the topic. In case I need to do anything here please be more specific.

Comment (sslavic) [reply CHANGELOG.md:8]: Guideline I was referring to is only for operators. It was shared in #news-dev via https://gigantic.slack.com/archives/C04TGHDEF/p1583408026036700
Work on enforcing it is covered in https://github.com/giantswarm/giantswarm/issues/9251

**Comment by paurosello** [comment]
Looks good to me, Stevo's questions are valid but don't have an answer

File: CHANGELOG.md (hunk 3 above)
Comment (xh3b4sd) [reply CHANGELOG.md:8]: I am still not sure what the question is or what I can do to help here.

Comment (omegas27) [reply CHANGELOG.md:8]: Thanks for the guidelines Stevo :)

Tim, I think Stevo just mentionned you to explain why and who is enforcing this separate PR guidelines and to give more context if needed.
He did a pretty good job to explain the rationale and challenges so I do not thing you need to help :)

Comment (kopiczko) [reply CHANGELOG.md:8]: > For that purpose probably even adding entry to Unreleased changelog update should be included in a separate PR from the one making the actual change, as reverting change commit should probably not revert changelog entry.

I think only release header should be added separately. Entries should be added with the feature/fix. When feature/fix is reverted then the entries will get removed. As long as something is not released this is what we want.

File: helm/coredns-app/values.yaml (hunk 2 above)
Comment (omegas27) [reply helm/coredns-app/values.yaml:28]: Still waiting for the rationale from @pipo02mix before merging :)

Comment (pipo02mix) [reply helm/coredns-app/values.yaml:28]: Sorry for delay. The reason was the customer who requested wanted to set sequential as default. but we can make it configurable and set out own default, no blocking

Comment (sslavic) [reply helm/coredns-app/values.yaml:28]: Thanks Fernando for heads up!

IIUC rather than keeping `forwardOptions` as required (when `forward` is used) and configuring our own default (even if aligned to upstream default at this moment), it might make more sense to modify the ConfigMap template to make `forwardOptions` optional so that `forwardOptions` default to upstream default automatically with every update to new upstream release; still we should keep `forwardOptions` as configurable for backward compatibility.

Comment (omegas27) [reply helm/coredns-app/values.yaml]: @sslavic I configured it as optional now :)
I will merge after you approve

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
...
+
+### Changed
+
+- Define default forwardOptions if we specify a forward in the chart.
```
Comment (sslavic) [inline CHANGELOG.md:12]:
```suggestion
- Make forward Options optional.
```

File: helm/coredns-app/values.yaml [hunk 5]
```diff
@@ -20,10 +20,12 @@ configmap:
   log: |
     denial
     error
-  # Uncoment forward and forwadOptions to enable DNS forward https://coredns.io/plugins/forward/
+
+  # Uncomment and define `forward` to enable and configure forwarding rules for CoreDNS forward plugin (for syntax see https://coredns.io/plugins/forward/)
+  # Default `forwardOptions` can be defined separately below; these are applied when forward plugin is enabled.
```
Comment (sslavic) [inline helm/coredns-app/values.yaml:25]:
```suggestion
```

**Comment by omegas27** [comment]
Tested multiple times on godsmack :rocket: 
Thanks for the really useful link @sslavic
